### DISCUSSION TITLE
Is  PERFORMANCE_TESTS required for developing new storage plugin?

### DISCUSSION THREAD

**Discussion by luxianlong** [JanusGraph/janusgraph#4594] [Q&A] [answered]
Hi, everyone, I'm developing a new storage plugin for JanusGraph based Microsoft Azure Table. I find that there're some tests marked as "PERFORMANCE_TESTS" which is disabled by default. I want to understand the design purpose of these tests. Are they required to pass for a new storage plugin? And if there is any performance number of existing storage plugin to compare with? 

Thanks!

**Comment by li-boxuan** [comment, marked as answer]
> Are they required to pass for a new storage plugin?

I wouldn't say so for any type of tests. Every storage plugin, including the officially supported ones, has some tests disabled due to their nature. Of course it will be great to pass as many tests as possible.

`JanusGraphConcurrentTest` is likely the most valuable one among these labeled as `PERFORMANCE_TESTS`.

> And if there is any performance number of existing storage plugin to compare with?

No. We do have some performance numbers though - every commit on main branch has a regression testing report, e.g. [this](https://github.com/JanusGraph/janusgraph/commit/2297fde2fbefcd8f983d4d7ac77fab014b548e5f#commitcomment-144758090). Those benchmarks run on a GitHub Action container, which can be unstable, so the perf numbers are just a reference.

**Reply by luxianlong** [reply]
Thanks @li-boxuan. It helps a lot.
